Assets/Packages/com.studio23.ss2.bettercursormanager/Runtime/Core/Controllers/CursorEventController.cs

```diff
@@ -0,0 +1,71 @@
+using System.Collections.Generic;
+using Studio23.SS2.BetterCursor.Interfaces;
+using UnityEngine;
+using UnityEngine.EventSystems;
+using UnityEngine.InputSystem;
+
+namespace Studio23.SS2.BetterCursor.Core
+{
+    public class CursorEventController : MonoBehaviour
+    {
+        private IHoverable _lastHoveredObject;
+
+        private LayerMask _layerMask;
+        private Camera _camera;
+        private List<RaycastResult> _raycastResults = new List<RaycastResult>();
+
+        internal void Initialize(LayerMask layerMask, Camera mainCamera)
+        {
+            _layerMask = layerMask;
+            _camera = mainCamera;
+        }
+
+        private void Update()
+        {
+            UpdateCollision();
+        }
+
+        private void UpdateCollision()
+        {
+            IHoverable currentHoveredObject = null;
+
+            if (BetterCursor.Instance.UiOnHoverEnabled)
+                currentHoveredObject = GetHoveredUIObject()?.GetComponent<IHoverable>();
+            else
+                currentHoveredObject = GetHoveredObject()?.GetComponent<IHoverable>();
+
+            if (currentHoveredObject == _lastHoveredObject) return;
+
+            if (_lastHoveredObject != currentHoveredObject)
+            {
+                _lastHoveredObject?.OnHoverExit();
+                _lastHoveredObject = currentHoveredObject;
+                _lastHoveredObject?.OnHoverEnter();
+            }
+        }
```

Comment:
Skip extra checks if UiHover is not enabled? Assuming that the collisions will only be checked when Hover is enabled.